### PR TITLE
US89636 - User can change banner image from existing menu

### PR DESCRIPTION
**Pull request by capajon**
let me know if you have any questions.  This mimics what my courses is doing very closely.  I thought about trying to share some code with a shared behavior but didn't think there was enough benefit for that vs the complication that comes with it.

I was also thinking about adding some tests for this (right now the tests are rather sparse) but I figured I would put it up for review first anyway.

### CODE REVIEW

File: src/d2l-image-banner-overlay.html [hunk 1]
```diff
@@ -175,6 +226,25 @@ <h1 id="courseName">[[_courseName]]</h1>
...
+				this._telemetryRequest.headers = {
+					'content-type': 'application/json'
+				};
+				this._telemetryRequest.body = JSON.stringify({
+					name: 'LaunchChangeImage',
+					ts: Math.round(Date.now() / 1000),
+					userId: this.userId,
+					tenantId: this.tenantId
+				});
+				this._telemetryRequest.generateRequest();
```
Comment (CodeBaboon) [inline src/d2l-image-banner-overlay.html]: Might be nice to refactor this telemetry stuff into its own method or even behavior, just something to clean up this function to be more like:

```
_launchCourseTileImageSelector: function() {
    this._sendTelemetryEvent({ name: 'LaunchChangeImage' });
    this.imageSelectorOpen = true;
    this.$['basic-image-selector-overlay'].open();
}
```

**Comment by jstefaniuk-d2l** [comment]
Hey, I'm glad to see this being worked on!

There's logic in the banner that causes the image to update when it's changed from a `my-courses` widget, it may be worth checking to see if the my-courses widget updates the course's image when changed from the banner, and if not ask someone if that's something that needs to be done as part of this story

**Comment by capajon** [comment]
@jstefaniuk-d2l That is something I was wondering about a bit and was asking my TPM about (but was okay with how things worked for now, especially being behind a feature flag).  So I don't quite get it, is the idea to make sure everyone sees the new image immediately?  I guess I am kind of surprised we have that requirement (wouldn't waiting until the page is refreshed be sufficient?)

**Comment by ryantmer** [comment]
It was a requirement in the My Courses widget when you changed the course image that the course tile would refresh - not sure if that would apply to the banner as well, though? (Seems like distinct functionality, although it does look great!)

File: src/d2l-image-banner-overlay.html (hunk 1 above)
Comment (capajon) [inline src/d2l-image-banner-overlay.html]: Sure, as a follow up I'll try creating a behavior for that (I might just keep it in this repo to start and then move it to our shared behaviors library and refactor the course widget after I know things are working well).

**Comment by capajon** [comment]
Sounds good.  I think that it is working in that way then (if you change the image through the image browser the banner updates immediately).  I'll confirm with my TPM though that it is what is expected (it is behind a feature flag for now).

**Comment by jstefaniuk-d2l** [comment]
Yeah I put the image spinner/updating logic in before leaving the project, so that should just work I think

As for updating my-courses when the banner updates, I don't have an opinion either way but it was a planned thing beforehand so I just wanted to check to see if people were aware of it (It may be out of scope)

**Pull request merged**